tests/integration/test_cloudformation.py

```diff
@@ -783,29 +856,49 @@ def test_cfn_handle_secretsmanager_secret(self):
         )
         self.assertIn('DeletedDate', rs)
 
-    def test_cfn_handle_firehose(self):
+    def test_cfn_handle_kinesis_firehose_resources(self):
         stack_name = 'stack-%s' % short_uid()
-        firehose_name = 'firehose-%s' % short_uid()
-        role_name = 'firehose-role-%s' % short_uid()
+        kinesis_stream_name = 'kinesis-stream-%s' % short_uid()
+        firehose_role_name = 'firehose-role-%s' % short_uid()
+        firehose_stream_name = 'firehose-stream-%s' % short_uid()
 
         cloudformation = aws_stack.connect_to_service('cloudformation')
-        params = [{'ParameterKey': 'DeliveryStreamName', 'ParameterValue': firehose_name}]
-        cloudformation.create_stack(StackName=stack_name, TemplateBody=TEST_TEMPLATE_12 % role_name, Parameters=params)
+        cloudformation.create_stack(
+            StackName=stack_name,
+            TemplateBody=TEST_TEMPLATE_12 % firehose_role_name,
+            Parameters=[
+                {'ParameterKey': 'KinesisStreamName', 'ParameterValue': kinesis_stream_name},
+                {'ParameterKey': 'DeliveryStreamName', 'ParameterValue': firehose_stream_name}
+            ]
+        )
 
         details = _await_stack_completion(stack_name)
 
         outputs = details.get('Outputs', [])
         self.assertEqual(len(outputs), 1)
 
+        kinesis_client = aws_stack.connect_to_service('kinesis')
         firehose_client = aws_stack.connect_to_service('firehose')
 
         rs = firehose_client.describe_delivery_stream(
-            DeliveryStreamName=firehose_name
+            DeliveryStreamName=firehose_stream_name
         )
-
-        self.assertEqual(firehose_name, rs['DeliveryStreamDescription']['DeliveryStreamName'])
         self.assertEqual(outputs[0]['OutputValue'], rs['DeliveryStreamDescription']['DeliveryStreamARN'])
+        self.assertEqual(firehose_stream_name, rs['DeliveryStreamDescription']['DeliveryStreamName'])
+
+        rs = kinesis_client.describe_stream(
+            StreamName=kinesis_stream_name
+        )
+        self.assertEqual(rs['StreamDescription']['StreamName'], kinesis_stream_name)
+
         cloudformation.delete_stack(StackName=stack_name)
+        time.sleep(2)
```

Comment:
We should try to avoid `sleep` with assumptions about the particular sleep duration, if possible. What are we waiting for here - is it the stack status changing to `DELETED`? Can we maybe use `retry` here with a short retry period to poll for the deletion status?